### PR TITLE
Add support for symlinks to target-shell

### PR DESCRIPTION
**Pull request by Horofic**
(DIS-1838)

### CODE REVIEW

File: dissect/target/loaders/tar.py [hunk 1]
```diff
@@ -47,7 +48,7 @@ def map(self, target: target.Target):
                     target.filesystems.add(vol)
 
                 volume = volumes["/"]
-                entry = TarFile(volume, member.name, member.name, self.tar)
+                entry = TarFile(volume, member.name, member, self.tar)
```
Comment (Schamper) [inline dissect/target/loaders/tar.py:51]: I believe the rest of the changes in this file have been superseded by https://github.com/fox-it/dissect.target/pull/112?

File: dissect/target/loaders/tar.py [hunk 2]
```diff
@@ -24,7 +25,7 @@ def __init__(self, path: Union[Path, str], **kwargs):
                 "Consider uncompressing the archive before passing the tar file to Dissect."
             )
 
-        self.tar = tarfile.open(path)
+        self.tar = tarfile.open(path, ignore_zeros=True)
```
Comment (Schamper) [inline dissect/target/loaders/tar.py:28]: Was this still necessary if you fixed it in `acquire` itself?

File: dissect/target/tools/shell.py [hunk 3]
```diff
@@ -504,17 +511,26 @@ def cmd_ls(self, args, stdout):
...
         if hasattr(entry, "deref") and ":" in name:
             # retrieve the ads from the NTFS entry
             ads_path = fsutil.join(fsutil.dirname(entry.path, separator=entry.fs.alt_separator), name)
             entry = entry.fs.get(ads_path)
 
-        stat = entry.lstat()
+        try:
+            stat = entry.stat()
+        except FileNotFoundError:  # Incase symlinks cant be found, due to them not being picked up by acquire
+            stat = entry.lstat()
```
Comment (Schamper) [inline dissect/target/tools/shell.py:525]: Shouldn't this remain `lstat`? The `ls` command doesn't show the stat information of the target but rather of the symlink file itself.

File: dissect/target/tools/shell.py [hunk 4]
```diff
@@ -545,9 +561,20 @@ def cmd_stat(self, args, stdout):
...
+        symlink = f"-> {path.readlink()}" if path.is_symlink() else ""
+
+        if path.is_symlink() and args.dereference:
+            s = path.stat()
+        elif path.is_symlink():
+            s = path.lstat()
+        elif args.dereference:
+            s = path.stat()
+        else:
+            s = path.stat()
```
Comment (Schamper) [inline dissect/target/tools/shell.py:573]: This looks very duplicate, shouldn't it still be `s = path.stat() if args.dereference else path.lstat()`?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/loaders/tar.py (hunk 1 above)
Comment (Horofic) [reply dissect/target/loaders/tar.py:51]: Looks like it! Merged those changes into this branch

File: dissect/target/tools/shell.py (hunk 3 above)
Comment (Horofic) [reply dissect/target/tools/shell.py:525]: Yep! Going to revert it back to just `entry.lstat()`

File: dissect/target/loaders/tar.py [hunk 5]
```diff
@@ -23,7 +23,9 @@ def __init__(self, path: Union[Path, str], **kwargs):
                 f"Tar file {path!r} is compressed, which will affect performance. "
                 "Consider uncompressing the archive before passing the tar file to Dissect."
             )
-
+        # Some issues exist with some tar entries, causing all entries afterwards to become invisible.
+        # Using `ignore_zeros=True` resolves this issue, but is a workaround.
+        # A propper solution should be implemented in the future
```
Comment (Schamper) [inline dissect/target/loaders/tar.py:28]:
```suggestion
```
Comment can be removed now.

File: dissect/target/tools/shell.py [hunk 6]
```diff
@@ -504,6 +511,7 @@ def cmd_ls(self, args, stdout):
     def print_extensive_file_stat(self, stdout, target_path: fsutil.TargetPath, name: str) -> None:
         """Print the file status."""
         entry = target_path.get()
+        symlink = ""
```
Comment (Schamper) [inline dissect/target/tools/shell.py:514]:
```suggestion
```

File: dissect/target/tools/shell.py [hunk 7]
```diff
@@ -512,9 +520,14 @@ def print_extensive_file_stat(self, stdout, target_path: fsutil.TargetPath, name
             entry = entry.fs.get(ads_path)
 
         stat = entry.lstat()
+
+        if entry.is_symlink():
+            symlink = f"-> {entry.readlink()}"
```
Comment (Schamper) [inline dissect/target/tools/shell.py:525]:
```suggestion
        symlink = f"-> {entry.readlink()}" if entry.is_symlink() else ""
```
You use a one liner elsewhere, might as well also use it here.

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Comment by Schamper** [comment]
Can you change the commit message to be "Add support for symlinks to target-shell"? Because I think in the end that's what this PR does.

**Pull request merged**